### PR TITLE
1.21.1 Fabric and NeoForge

### PR DESCRIPTION
**Pull request by ChrysanthCow**
- Abstracted over 1.21 specifics in ApathyPlusMinecraft for each version.
- Now using ResourceKeys for MobEffect abstraction as 1.21 primarily uses holders for them.
- Fabric is pretty much the same as before.
- NeoForge now uses Fabric Mixin, so we can get rid of the crusty mixin from it.
- IExtensionPoint.DisplayTest has been removed from NeoForge. There is no replacement for it.

### CODE REVIEW

File: common-1.21.1/src/main/java/agency/highlysuspect/apathy/Apathy121.java [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+public abstract class Apathy121 extends ApathyPlusMinecraft {
+	public static Apathy121 instance121;
+	
+	public Apathy121() {
+		if(instance121 == null) {
+			instance121 = this;
+		} else {
+			IllegalStateException e = new IllegalStateException("Apathy 1.20 instantiated twice!");
+			log.error("Apathy 1.20 instantiated twice!", e);
```
Comment (quat1024) [inline common-1.21.1/src/main/java/agency/highlysuspect/apathy/Apathy121.java]: log ^^

File: neoforge-1.21.1/src/main/java/agency/highlysuspect/apathy/platform/neoforge/LegacyToTomlUpgrader.java
Comment (quat1024) [inline neoforge-1.21.1/src/main/java/agency/highlysuspect/apathy/platform/neoforge/LegacyToTomlUpgrader.java:1]: I think we can delete this file now

File: neoforge-1.21.1/src/main/java/agency/highlysuspect/apathy/platform/neoforge/mixin/EntityGetterMixin.java
Comment (quat1024) [inline neoforge-1.21.1/src/main/java/agency/highlysuspect/apathy/platform/neoforge/mixin/EntityGetterMixin.java:1]: Now that Neoforge has the good version of mixin, could this be brought into common-1.21.1's pile of mixins? That way it wouldn't be duplicated

File: neoforge-1.21.1/src/main/resources/META-INF/neoforge.mods.toml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+license="LGPL-3.0-or-later"
+issueTrackerURL="https://github.com/quat1024/apathy/issues"
+
+[[mods]]
+modId="apathy"
+version="${version}"
+displayName="Apathy"
+displayURL="https://github.com/quat1024/apathy/"
+logoFile="apathy-icon-128.png"
+credits="whizball1 for Apathetic Mobs, jaredl0008 and Darkhax for MultiLoader-Template"
```
Comment (quat1024) [inline neoforge-1.21.1/src/main/resources/META-INF/neoforge.mods.toml]: You  can add yourself to the credits if you want

**Review by quat1024 — COMMENTED** [review]
thanks so much💖

File: neoforge-1.21.1/src/main/resources/META-INF/neoforge.mods.toml (hunk 2 above)
Comment (ChrysanthCow) [reply neoforge-1.21.1/src/main/resources/META-INF/neoforge.mods.toml]: I forgot to do that on NeoForge, whoops. Did it on Fabric tho

**Pull request merged**